python_modules/libraries/dagster-aws/dagster_aws/ecs/client.py

```diff
@@ -1,259 +1,138 @@
-
-    Args:
-        region_name (Optional[str]):  which AWS region you're running on
-        key_id (Optional[str]): the AWS access key ID to use
-        access_key (Optional[str]): the AWS access key going with that key_id
-        launch_type (Optional[str]): whether to use EC2 or FARGATE for the task
-        grab_logs (Optional[bool]):  whether to pull down ECS logs
-        starter_client (boto3.client): starter ecs client, used for mocking
-        starter_log_client (boto3.client): starter log client, used for mocking
-
+class ECSError(Exception):
+    pass
 
-    TODO: find ECS specific logs, if they exist -- right now logs are stdout and stderr
-    """
-
-    def __init__(
-        self,
-        region_name="us-east-2",
-        key_id="",
-        access_key="",
-        launch_type="FARGATE",
-        starter_client=None,
-        starter_log_client=None,
-        grab_logs=True,
-    ):
-        self.key_id = check.opt_str_param(key_id, "key_id", "")
-        self.region_name = check.str_param(region_name, "region_name")
-        self.access_key = check.opt_str_param(access_key, "access_key")
-        self.launch_type = check.str_param(launch_type, "launch_type")
-        self.account_id = "123412341234"
-        self.grab_logs = check.bool_param(grab_logs, "grab_logs")
-        self.task_definition_arn = None
-        self.tasks = []
-        self.tasks_done = []
-        self.offset = -1
-        self.container_name = None
-        self.ecs_logs = dict()
-        self.logs_messages = dict()
-        self.warnings = []
-        self.cluster = None
-        if starter_client is None:
-            self.ecs_client = self.make_client("ecs")
-            self.account_id = boto3.client("sts").get_caller_identity().get("Account")
-        else:
-            self.ecs_client = starter_client
-        if starter_log_client is None:
-            self.log_client = self.make_client("logs")
-        else:
-            self.log_client = starter_log_client
-        self.set_cluster()
 
-    def make_client(self, client_type, **kwargs):
-        """
-        constructs an ECS client object with the given params
+class ECSTimeout(ECSError):
+    pass
 
-        Args:
-            client_type (str): which ECS client you're trying to produce
-            **kwargs: any add'l keyword params
-        Returns:
-            client: the constructed client object
-        """
-        if self.key_id != "":
-            clients = boto3.client(
-                client_type,
-                aws_access_key_id=self.key_id,
-                aws_secret_access_key=self.access_key,
-                region_name=self.region_name,
-                **kwargs,
-            )
-        else:
-            clients = boto3.client(client_type, region_name=self.region_name, **kwargs)
 
-        return clients
+class ECSClient:
+    def __init__(self, client, cluster, subnets, polling_interval=5, max_polls=120):
```

Comment:
This departs from the ECS API a bit; when running a Fargate task, you pass network configuration when you call `run_task`. Given that this client is intentionally more narrowly scoped than the ECS API, I think it's appropriate to set the same network configuration for all tasks that the client will be running. But as I iterate on this, I might find reason to pull this back into the `run_task` method instead of including it in the initializer.

Reply:
sounds good 👍